File: Striver Important Interview Questions/Day 2 - Arrays Part 2/rotate-image.cpp

// https://leetcode.com/problems/rotate-image/description/

class Solution {
public:
    void rotate(vector<vector<int>>& matrix) {
        int rowLen = matrix.size();
        int colLen = matrix[0].size();

        for(int i = 0; i < rowLen; i++){
            for(int j = 0; j < i; j++){
                swap(matrix[i][j], matrix[j][i]);
            }
        }

        for(int i = 0; i < rowLen; i++){
            reverse(matrix[i].begin(), matrix[i].end());
        }
    }
};

File: Neetcode-150-js/Math & Geometry/rotate-image.js

// https://leetcode.com/problems/rotate-image/

/**
 * @param {number[][]} matrix
 * @return {void} Do not return anything, modify matrix in-place instead.
 */
var rotate = function(matrix) {
    let len = matrix.length;
    if(len === 0) return;

    for(let i = 0; i < len; i++){
        for(let j = 0; j < i; j++){
            [matrix[i][j], matrix[j][i]] = [matrix[j][i], matrix[i][j]];
        }
    }

    for(let i = 0; i < len; i++){
        matrix[i].reverse();
    }

    return;
};